packages/composite-checkout/src/components/checkout-submit-button.tsx

```diff
@@ -1,21 +1,79 @@
 /**
  * External dependencies
  */
-import React from 'react';
+import React, { useCallback } from 'react';
+import debugFactory from 'debug';
+import { useI18n } from '@automattic/react-i18n';
 
 /**
  * Internal dependencies
  */
 import joinClasses from '../lib/join-classes';
-import { usePaymentMethod } from '../public-api';
+import { usePaymentMethod, usePaymentProcessors, useTransactionStatus } from '../public-api';
+import { PaymentProcessorResponse } from '../types';
+
+const debug = debugFactory( 'composite-checkout:checkout-submit-button' );
 
 export default function CheckoutSubmitButton( {
 	className,
 	disabled,
 }: {
 	className?: string;
 	disabled?: boolean;
-} ) {
+} ): JSX.Element | null {
+	const {
+		setTransactionComplete,
+		setTransactionRedirecting,
+		setTransactionError,
+		setTransactionPending,
+	} = useTransactionStatus();
+	const paymentProcessors = usePaymentProcessors();
+	const { __ } = useI18n();
+	const redirectErrorMessage = __(
+		'There was an error redirecting you to the payment partner. Please try again or contact support.'
```

Comment:
:information_source: String reuse speeds up translation and improves consistency. The following string might make a good alternative and has already been translated **19** times:
`translate( 'An error occurred while redirecting to the payment partner. Please try again or contact support.' )` *ES Score: 7*